pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix

```
@@ -0,0 +1,144 @@
+{ lib
+, fetchurl
+, runCommand
+
+  # script interpteers
+, bash
+, jdk
+, perl
+, python3
+, ruby
+, snobol4
+, tk
+
+  # compiled binaries
+, texliveBinaries
+  # fixed hashes
+, texliveFixedHashes
+  # mirrors
+, texliveMirrors
```

The parameters are self-explaining enough.

```suggestion
, texliveBinaries
, texliveFixedHashes
, texliveMirrors
```